src/services/distributions.ts

```diff
@@ -19,21 +19,25 @@ const schemaConstructor = Joi.object({
 export interface QueryOptions {
   source: IRI;
   query: string;
+  timeoutMs: number;
 }
 
 const schemaQuery = Joi.object({
   source: Joi.object().required(),
   query: Joi.string().required(),
+  timeoutMs: Joi.number().required(),
```

Comment:
We could add the timeout options to the Joi object, e.g.:

`timeoutMs: Joi.number().integer().min(1000).max(10000).default(10000)`